repos/system_upgrade/common/actors/checkldconf/libraries/checkldsoconfiguration.py

```diff
@@ -0,0 +1,98 @@
+import glob
+import os
+
+from leapp import reporting
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import InstalledRedHatSignedRPM
+
+LD_SO_CONF_DIR = '/etc/ld.so.conf.d'
+LD_SO_CONF_MAIN = '/etc/ld.so.conf'
+LD_SO_CONF_DEFAULT_INCLUDE = 'ld.so.conf.d/*.conf'
+LIST_FORMAT_PREFIX = '\n    - '
+
+
+def _read_file(file_path):
+    with open(file_path, 'r') as fd:
+        return fd.readlines()
+
+
+def _report_custom_ld_so_configuration(summary):
+    reporting.create_report([
+        reporting.Title(
+            'Third party libraries linked with ld.so.conf are not supported for the in-place upgrade.'
+        ),
+        reporting.Summary(summary),
+        reporting.Remediation('Remove the custom ld.so configuration.'),
+        reporting.RelatedResource('file', '/etc/ld.so.conf'),
+        reporting.RelatedResource('directory', '/etc/ld.so.conf.d'),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Groups([reporting.Groups.OS_FACTS, reporting.Groups.INHIBITOR]),
+    ])
+
+
+def _is_included_ld_so_config_custom(config_path):
```

Comment:
I will maybe look into the `rpm -V` used for verification and maybe incorporate it in. This is a reasonable concern we should maybe address.

Reply:
I further enhanced the detection of custom configuration files when they belong to a RH-signed RPM by running a `rpm -Va` command to detect any form of modification. Please let me know what you think about this approach.

Reply:
I think this approach is better, even though it might produce false positives if, for example, a comment was added to a config file. Such change would get flagged as customization even though it doesn't functionally change the config. Since the produced report isn't a inhibitor anymore I think it's okay to not consider such an edge case. Also to truly say whether the config change was functional we would have to parse diffs which is IMHO an overkill.

Reply:
@pirat89 What do you think about this?

Reply:
Tsorry for the late response. yes, we should check whether we speak about configuration files made by RH signed packages or not as we expected in case of RH signed packages (if configs are not modified) that these are ok. However, we need to check the behaviour in case of RH SCL.